src/Task/Generic/CreateSymlinksTask.php

```diff
@@ -57,7 +57,7 @@ public function execute(Node $node, Application $application, Deployment $deploy
             'cd ' . $baseDirectory
         ];
         foreach ($options['symlinks'] as $linkPath => $sourcePath) {
-            $commands[] = 'ln -s ' . $sourcePath . ' ' . $linkPath;
+            $commands[] = '[ -e ' . $linkPath . ' ] && rm ' . $linkPath . ' && ln -s ' . $sourcePath . ' ' . $linkPath;
```

Comment:
As there is a new release folder each deployment, it shouldn't be necessary to delete a symlink.

Reply:
Hi, I needed this, because the initialization for Neos/Flow creates a Web/_Resources/.htaccess with settings that are not always available with all servers. So I needed to overwrite this file with one in my shared folder. Of course you could argue, that we should change Neos, but I believe that this may happen with others files too.

Reply:
mhmm i also always alter the .htaccess during deployment, most of the time with `sed`

Reply:
because what you typically want to do is:

```php
$workflow->defineTask(
        'FixHtaccess',
        ShellTask::class,
        array(
            'command' => 'cd {releasePath} ; sed -i "s/RewriteBase \/.*/RewriteBase \/' . $GLOBALS['rewrite_path'] . '\/ \n SetEnv FLOW_CONTEXT ' . addcslashes($GLOBALS['context'], '/\\.') . ' /g" {releasePath}/Web/.htaccess'
        )
    );
```

so we normally have a default for ddev in our git and do the replacement during deployment.

Reply:
We can find several workarounds, but if you configure a symlink, do you expect the file that you ask for or the existing one. 
Of course there are other ways, but what would be the correct output of this function?

Reply:
i would expect this action to be atomic, that means, i would like to have the link deleted and recreated

Reply:
But it's not save with this check. It will delete any file and replace it with a symlink in the new release. It's dangerous de facto.
But we could add an option to that task to force delete the link target.
Or you add a DeleteHtaccessTask before CreateSymlinksTask.

Reply:
Ahh now i get it ... it's the case if the .htaccess already exists, e.g. via git ... mhmm, as it comes from git there should be a warning ...

using .htaccess with symlinks is anyway also likely to break on many hosters

Reply:
The most important thing is that it must not be the default to silently delete files while creating symlinks.

Reply:
agree